Incorporate comments from Stuart's review (#56)
tfpauly/draft-pauly-adaptive-dns-privacy: draft-pauly-adaptive-dns-privacy.md

Clients MUST ignore any DoH server URI that was not retrieved from a
DNSSEC-signed record that was validated by the client {{!RFC4033}}.

Whenever a client resolves a name for which it does not already have a Designated DoH Server,
it SHOULD try to determine the Designated DoH Server by sending a query for the appropriate

> SVCB record. If there is no DoH server designated for the name or zone, signalled either

Author: Changed the wording to just say its for the same name!

least one query that is targeted at the server through a proxy before sending direct queries
to the server.

> Designated DoH Servers are expected to act both as Oblivious Proxies and Oblivious Targets

Reviewer: ```suggestion
Designated DoH Servers are expected to act both as Oblivious Proxies and as Oblivious Targets
```

to ensure that clients have sufficient options for preserving privacy using Oblivious DoH.
Oblivious Targets are expected to act as Oblivious Proxies to ensure that no Oblivious DoH server
can act as only a target (thus being able to see patterns in name resolution, which might have

> value to a resolver) and require other servers to take on a disproportionate load of proxying.

Reviewer 2: We should probably also recommend some client behavior if certain targets are not willing to proxy. Perhaps, "Clients SHOULD stop using Oblivious Targets in the absence of proxy support"?

Author: I think this is already covered by the beginning of the section, where the clients MUST whitelist the servers by validating that they can both be the proxy and target. You should never use a server as a target if it isn't a proxy too.

When a Designated DoH Server is discovered, clients SHOULD also check to see
if this server provides an extended configuration in the form of a Web PvD ({{configuration}}).
To do this, the client performs a GET request to the DoH URI, indicating that it accepts
a media type of “application/pvd+json” {{!I-D.ietf-intarea-provisioning-domains}}. When requesting

> the PvD information, the query and fragment components of the requested path SHOULD be left

Author: Reworded to just say it is left empty; it's not really a normative thing, it's just that I'm pointing out that the query string doesn't have the same meaning as DoH here.

> empty. Note that this is different from GET requests for the “application/dns-message” type,

Reviewer: ```suggestion
empty. Note that this is different from a GET request for the “application/dns-message” type,
```

When establishing a secure connection to a certain hostname, clients need
to first determine which resolver configuration ought to be used for DNS resolution.

Several of the steps outlined in this algorithm take into account the success or failure
of name resolution. Failure can be indicated either by a DNS response, such as SERVFAIL

> NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,

Reviewer: ```suggestion
or NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,
```

> or NXDOMAIN, or by a connection-level failure, such as a TCP reset, and TLS handshake failure,

Reviewer 2: ```suggestion
or NXDOMAIN, or by a connection-level failure, such as a TCP reset, TLS handshake failure,
```

# Performance Considerations

One of the challenges of using non-local DNS servers (such as cloud-based DoH servers)
is that recursive queries made by these servers will originate from an IP address that
is not necessarily geographically related to the client. Many DNS servers make assumptions
about the geographic locality of clients to their recursive resolvers to optimize answers.
To avoid this problem, the client's subnet can be forwarded to the authoritative server
by the recursive using the EDNS0 Client Subnet feature. Oblvious DoH discourages this practice
for privacy reasons. However, sharing this subnet, while detrimental to privacy, can result in
better targeted DNS resolutions.

> Adaptive DNS splits DoH queries up into two sets: those made to Designated DoH Servers,

Reviewer: ```suggestion
Adaptive DNS splits DoH queries into two sets: those made to Designated DoH Servers,
```
